### PR TITLE
Add Classes

### PR DESCRIPTION
**Pull request by yosefHesham**
### Changes Made:
---

- Add person class
- Add student class extends person.
- Add teacher class extends person.

### CODE REVIEW

**Review by whiz25 — CHANGES_REQUESTED** [review]
Hi @yosefHesham, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has failing linters checks. Please fix those errors and resubmit

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

### Martin Afani
#### Slack: whiz25

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  def initialize(age, name = 'Unkown', _parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+  end
```
Comment (KDlamini) [inline person.rb]: Kindly add the `@@parent_permission` variable inside the initialize method as well...

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def can_use_services?
+    of_age?
+  end
+
```
Comment (KDlamini) [inline person.rb]: The method `can_use_services?` should return true if a person is of age or **if they have permission from parents**.

- Kindly include the `@parent_permission` variable in your return statement.

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Student < Person
+  def initialize(age, name, classroom)
+    super(age, name)
+    @classroom = classroom
+  end
```
Comment (KDlamini) [inline student.rb]: Kindly include the `parent_permission` parameter in your `initialize` and `super` constructors.

- Kindly make sure that the default value for `name` is  "Unknown" and the default value for `parent_permission`  is set to "true".

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Teacher < Person
+  def initialize(age, name, specilization)
+    super(age, name)
+    @specilization = specilization
+  end
```
Comment (KDlamini) [inline teacher.rb]: Same here, kindly include the `parent_permission` parameter in your `initialize` and `super` constructors and make sure that the default value for `name` is  "Unknown" and the default value for `parent_permission`  is set to "true".

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @yosefHesham 🙋‍♀️,  

Great work so far! 💯 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 

✔️  Pull Request is professional
✔️  There is an individual `.rb` file for each class (`Person`, `Student`, and `Teacher` )
✔️  No linter errors

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — COMMENTED** [review]
Hi @yosefHesham , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive PR
:heavy_check_mark:  All linters checks passed 
:heavy_check_mark: lmplemented all project requirements 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**